Update install notes (#749)
CrayLabs/SmartSim: doc/installation_instructions/troubleshooting/custom_backends.rst

Custom ML backends
------------------

The ML backends (Torch, ONNX Runtime, and Tensorflow) and their associated
python packages have different versions and indices that can be supported based

> on the intended device (CPU, ROCM, CUDA-11, or CUDA-12). The officially

Reviewer: ROCM -> ROCm?

supported backends are stored in JSON files within the
``smartsim/_core/_install/configs/mlpackages`` directory.

> If you need to define a different version of the backend and/or the packages, we

Reviewer 2: ```suggestion
To customize the version of a backend or package, we
```

> recommend that you copy one of the JSON files (for example the one at the end of

Reviewer 2: ```suggestion
recommend that you use the configuration shipped with SmartSim as a template.

Copy the file and update as required. Afterward, run ``smart build --config-dir <path>/<to>/<file>.json`` to tell SmartSim to use your custom configuration(s).```

Custom ML backends
------------------

> The ML backends (Torch, ONNX Runtime, and Tensorflow) and their associated

Reviewer: supported based on the intended device (CPU, ROCm, CUDA-11, or CUDA-12) - I think the only device in the list is CPU right? Maybe say device or platform
